### PR TITLE
Fix undefined properties in where

### PR DESCRIPTION
**Pull request by loay**
### Description

backport of https://github.com/strongloop/loopback-datasource-juggler/pull/1416

### Checklist

- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style
  guide](http://loopback.io/doc/en/contrib/style-guide.html)

### CODE REVIEW

**Comment by virkt25** [comment]
Not just mongo, cloudant, soap and 2 others are failing.

**Comment by loay** [comment]
@slnode test please

**Comment by loay** [comment]
@slnode test please

File: lib/dao.js [hunk 1]
```diff
@@ -2187,7 +2187,7 @@ DataAccessObject.destroyAll = function destroyAll(where, options, cb) {
         var handleUndefined =  Model._getSetting('normalizeUndefinedInQuery');
         // alter configuration of how removeUndefined handles undefined values
         where = removeUndefined(where, handleUndefined);
-        where = Model._coerce(where, options);
+        where = Model._coerce(where);
```
Comment (jannyHou) [inline lib/dao.js]: hmm...why some functions accept `options` but these two not?

Comment (loay) [inline lib/dao.js]: The `options` object are empty, therefore the test case failed in 2 occurrences.. 
once in juggler
```
 should count using date string
```
and once in Mongo 

```
mongodb default id type should be able to delete by string id
```
Where the `option` variable is not even affection the current fix as it is not needed..
The reason why , it seems that once the `Option` object is empty, it causes  an assertion failure, then the test case fail.

**Comment by loay** [comment]
PR cannot be backported to 2.x for the simple reason that `._coerce` function signature has changed in 3.x, therefore it is not compatible.

**Pull request closed without merging**